tests/testthat/test_kpca.R

```diff
@@ -74,3 +74,45 @@ test_that('No kPCA comps', {
     tibble::tibble(terms = paste0("X", 2:6), id = "")
   )
 })
+
+
+test_that('keep_original_cols works', {
+
+  skip_if_not_installed("dimRed")
+  skip_if_not_installed("kernlab")
+
+  kpca_rec <- rec %>%
+    step_kpca(X2, X3, X4, X5, X6, id = "", keep_original_cols = TRUE)
+
+  kpca_trained <- prep(kpca_rec, training = tr_dat, verbose = FALSE)
+
+  pca_pred <- bake(kpca_trained, new_data = te_dat, all_predictors())
+
+  expect_equal(
+    colnames(pca_pred),
+    c("X2", "X3", "X4", "X5", "X6",
+      "kPC1", "kPC2", "kPC3", "kPC4", "kPC5")
+  )
+
+})
+
+test_that('can prep recipes with no keep_original_cols', {
+  skip_if_not_installed("dimRed")
+  skip_if_not_installed("kernlab")
+
+  kpca_rec <- rec %>%
+    step_kpca(X2, X3, X4, X5, X6, id = "")
+
+  kpca_rec$steps[[1]]$keep_original_cols <- NULL
+
+  expect_warning(
+    kpca_trained <- prep(kpca_rec, training = tr_dat, verbose = FALSE),
+    "'keep_original_cols' was added to"
+  )
+
+  expect_error(
+    pca_pred <- bake(kpca_trained, new_data = te_dat, all_predictors()),
+    NA
+  )
```

Comment:
Should these be `expect_snapshot()` now that we have testthat e3?

Reply:
They don't generate any output so I don't think so, right?

Reply:
`expect_snapshot()` capture the warnings and errors quite nicely, Davis used it several times in his e3 PR

Reply:
So it will capture an empty snapshot and we'll test for an empty snapshot in the future?

Reply:
Sorry if I'm being dense; I just haven't seen it used that way before 🙈

Reply:
ah, apologies, I overlooked the `NA` here :see_no_evil: